diff --git a/moviepy/audio/io/readers.py b/moviepy/audio/io/readers.py
index 991e60b..ffed1c4 100644
--- a/moviepy/audio/io/readers.py
+++ b/moviepy/audio/io/readers.py
@@ -93,6 +93,7 @@ class FFMPEG_AudioReader:
         if os.name == "nt":
             popen_params["creationflags"] = 0x08000000
 
+        # print('popen cmd={}, params={}'.format(cmd, popen_params))
         self.proc = sp.Popen( cmd, **popen_params)
 
         self.pos = np.round(self.fps*starttime)
@@ -109,8 +110,22 @@ class FFMPEG_AudioReader:
     def read_chunk(self,chunksize):
         # chunksize is not being autoconverted from float to int
         chunksize = int(round(chunksize))
+        '''
+        print('num channels: {}, chunk size: {}, num bytes: {}'.format(
+            self.nchannels, chunksize, self.nbytes))
+        '''
         L = self.nchannels*chunksize*self.nbytes
-        s = self.proc.stdout.read(L)
+        # print('Reading {} bytes from pipe stdout'.format(L))
+        # Modified to avoid read blocking
+        # Kill read after 60 seconds
+        from threading import Timer
+        kill = lambda process: process.kill()
+        my_timer = Timer(60, kill, [self.proc])
+        try:
+            my_timer.start()
+            s = self.proc.stdout.read(L)
+        finally:
+            my_timer.cancel()
         dt = {1: 'int8',2:'int16',4:'int32'}[self.nbytes]
         if hasattr(np, 'frombuffer'):
             result = np.frombuffer(s, dtype=dt)
